### ISSUE TITLE
Accessibility Problem with Manufacturers, Music Genres, and Record Companies Dropdowns

### ISSUE THREAD

**Issue by dbltoe** [zencart/zencart#4482]
I'm working on trying to get the Bootstrap template as close to ADA/WCAG as possible.  These things are not specific to Bootstrap but do show up in my testing.

Accessibility includes the requirement for someone to navigate a site with just a keyboard even if a virtual keyboard.  When using onChange such as these boxes, the keyboard must be able to make selections using the arrow keys WITHOUT moving away from the selection box.

In all three boxes, clicking on the Please Select, then using the down arrow immediately moves the browser to that selection.

If the customer tries to get to the Rock Music Genre, they cannot with a keyboard.  Same with Manufacturers.  GT Interactive would never be reached.

Right now, Record Companies only has one selection in the Demo but, a current client has hundreds to add.  Thus the reason for discovery.

index.php?main_page=product_info&cPath=25&products_id=100 is an example of the keyboard arrows working fine with the dropdown of attribute selection.

Not being a JavaScript person, I'm hoping that simply changing the event from onChange to onClick could be the solution.

**Comment by mc12345678** [comment]
For clarity, when you say "Dropdown" you do mean a selection list that is initially "compressed" but then can either be expanded by pressing the space bar or the dropdown arrow ***or*** do you mean how ZC defaults in a new store where a maximum list setting has been entered so that the element no longer 'appears' as a dropdown and no longer operates the way considered?

If I for example set: Configuration->Maximum Values->Manufacturers List - Scroll Box Size/Style to a 1, then when I have the dropdown (which truly is a dropdown that point and not a list) if I press the spacebar while it is "highlighted" then it responds as expected allowing movement to/from a selection without reloading the page. 

But in a default install, the options are displayed as a list not as a dropdown. Thus handling a list is different and yes, would likely want to use different javascript at that point. Changing the action to an onclick will only "select" the item and redirect the page when the mouse is used to click on an item in the list. There is no keyboard shortcut at that point which then wouldn't resolve the issue other than basically to display the list on each page and not provide navigation by keyboard. 

So seems, for compliance, should use a value of 1 (or 0 possibly) in the associated list identifier for these select and change options. I'm not seeing a "simple" one-liner change to support changing the behaviour. Now, if added a select button or something similar, then there certainly more options made available. Note, I am not saying that something else isn't possible, I did try applying the OnClick action while having a "sized" select element and the only time that navigation occurred was if I actually used a mouse to click the element.

As far as the dropdown associated with the attributes, that was one of the triggers to identify the difference between what was described and what was displayed. Attribute dropdowns tend to use a value of 1 instead of a listing.

Settings associated (Default for each is 3):
Manufacturers List - Scroll Box Size/Style
Music Genre List - Scroll Box Size/Style
Record Company List - Scroll Box Size/Style

**Comment by scottcwilson** [comment]
I am hopeful that we can get rid of Music Genres and Record Companies so please don't spend a ton of time on those.

**Comment by lat9** [comment]
I believe that this issue can be closed, based on the changes currently merged for the above-mentioned issue.  @dbltoe?

**Comment by dbltoe** [comment]
There is little more we can do with the sideboxes even though ANDI doesn't like the Disabled Go buttons on each.  THANX to you for your work on that.
There is a new item that has shown up on the main page.  Specifically in the footer's siteinfoLegal.
Should I make that a different issue?
If so, this issue can be closed and I will talk about the non-existant, non-titled link to the store's main page.  Nothing between the a tags and not title.

**Comment by dbltoe** [comment]
Just noticed that I have not added the store to my latest test site.  Will add that and see if the "new" concern still exists.
